experiments/AMIP/modular/coupler_driver_modular.jl

```diff
@@ -91,6 +93,19 @@ if isinteractive()
     parsed_args["monthly_checkpoint"] = true
 end
 
+config_dict = YAML.load_file(parsed_args["config_file"])
+atmos_config = if !isnothing(config_dict)
+    CA.override_default_config(config_dict)
+elseif !isnothing(parsed_args["config_file"])
+    CA.override_default_config(parsed_args["config_file"])
+else
+    @info "Using Atmos default configuration"
+    CA.default_config_dict()
+end
+
+config_dict = merge(config_dict, atmos_config)
+parsed_args = merge(parsed_args, config_dict)
```

Comment:
please comment that `config_dict` appends to and overwrites `parsed_args`.

Reply:
In fact, the `isinteractive()` loop is being overwritten right now. I think what you want is this sequence:

```
config_dict = YAML.load_file(parsed_args["config_file"])
config_dict = merge(config_dict, parsed_args)
atmos_config = if !isnothing(config_dict)
    CA.override_default_config(config_dict)
elseif !isnothing(parsed_args["config_file"])
    CA.override_default_config(parsed_args["config_file"])
else
    @info "Using Atmos default configuration"
    CA.default_config_dict()
end
parsed_args = merge(config_dict, atmos_config)
```

Reply:
(with the atmos-specific if statement moved to `climaatmos_init.jl`, and the last line can remain in the driver)

Reply:
I'm not sure I entirely follow this bit:

> with the atmos-specific if statement moved to climaatmos_init.jl

since we call `atmos_init` much later in the driver, and we need to make sure we override the default Atmos config (or leave the default one, in case no config file is specified) here in the driver

Reply:
I am testing this interactively (by running `julia -i`) and it works just fine. See this for example:

```
valeria@work-xps:~/CliMA/ClimaCoupler.jl$ julia -i --color=yes --project=experiments/AMIP/modular experiments/AMIP/modular/coupler_driver_modular.jl --config_file config/model_configs/slabplanet_default.yml 
WARNING: could not import Interfacer.update_sim! into Main
[ Info: /home/valeria/CliMA/ClimaCoupler.jl/experiments/AMIP/modular/output/slabplanet/slabplanet_default
[ Info: Dict{Any, Any}("test_edmf_consistency" => false, "surface_thermo_state_type" => "GCMSurfaceThermoState", "topo_smoothing" => false, "hyperdiff" => "ClimaHyperdiffusion", "dt" => "200secs", "output_dir" => nothing, "prognostic_tke" => false, "imex_edmf_gm" => false, "use_newton_rtol" => false, "device" => "auto", "t_end" => "10days", "dz_top" => 5000.0, "C_E" => 0.0044, "y_elem" => 6, "target_job" => nothing, "z_stretch" => true, "bubble" => true, "ode_algo" => "ARS343", "max_newton_iters_ode" => 1, "start_date" => "19790101", "imex_edmf_turbconv" => false, "forcing" => nothing, "edmfx_nh_pressure" => false, "monthly_checkpoint" => true, "prognostic_surface" => "false", "test_dycore_consistency" => false, "moist" => "equil", "rad" => "gray", "energy_name" => "rhoe", "dt_save_to_disk" => "Inf", "rayleigh_sponge" => false, "dt_save_restart" => "Inf", "energy_check" => true, "initial_condition" => "DecayingProfile", "krylov_rtol" => 0.1, "divergence_damping_factor" => 1.0, "perf_mode" => "PerfStandard", "zd_rayleigh" => 15000.0, "eisenstat_walker_forcing_alpha" => 2.0, "surface_setup" => "PrescribedSurface", "mono_surface" => true, "edmf_coriolis" => nothing, "smoothing_order" => 3, "idealized_h2o" => false, "perturb_initstate" => true, "turbconv_case" => nothing, "jvp_step_adjustment" => 1.0, "anim" => true, "discrete_hydrostatic_balance" => false, "entr_coeff" => 1.0, "dz_bottom" => 500.0, "trunc_stack_traces" => true, "h_elem" => 4, "restart_t" => 0, "turb_flux_partition" => "CombinedStateFluxes", "kappa_4" => 2.0e17, "advection_test" => false, "dt_cpl" => 200, "z_max" => 30000.0, "apply_limiter" => true, "precip_model" => "0M", "topography" => "NoWarp", "reference_job_id" => nothing, "perf_summary" => false, "viscous_sponge" => false, "config_file" => "config/model_configs/slabplanet_default.yml", "surface_temperature" => "ZonallySymmetric", "edmfx_sgs_flux" => false, "job_id" => "slabplanet_default", "density_upwinding" => "none", "orographic_gravity_wave" => nothing, "dt_rad" => "6hours", "run_name" => "slabplanet_default", "zd_viscous" => 15000.0, "edmfx_upwinding" => "none", "debugging_tc" => false, "tracer_upwinding" => "none", "edmfx_entr_detr" => false, "nh_poly" => 3, "y_max" => 300000.0, "non_orographic_gravity_wave" => false, "use_reference_state" => true, "coupled" => true, "config" => "sphere", "energy_upwinding" => "none", "FLOAT_TYPE" => "Float64", "split_ode" => true, "regression_test" => false, "check_conservation" => false, "restart_dir" => "unspecified", "ls_adv" => nothing, "x_max" => 300000.0, "z_elem" => 10, "newton_rtol" => 1.0e-5, "fps" => 5, "mode_name" => "slabplanet", "turbconv" => nothing, "x_elem" => 6, "vert_diff" => "true", "idealized_clouds" => false, "detr_coeff" => 0.001, "use_krylov_method" => false, "subsidence" => nothing, "use_dynamic_krylov_rtol" => false, "idealized_insolation" => true, "alpha_rayleigh_w" => 1.0, "alpha_rayleigh_uh" => 0.0001, "kappa_2_sponge" => 1.0e6, "toml" => Any[], "post_process" => true, "dt_save_to_sol" => "9days", "albedo_from_file" => true)
WARNING: could not import FieldExchanger.get_thermo_params into Main
WARNING: could not import Interfacer.get_model_state_vector into Main
[ Info: Running on CPUSingleThreaded.
[ Info: Setting up single-process ClimaAtmos run
[ Info: Running ClimaCore in unthreaded mode.
┌ Info: AtmosModel: 
│ 
│                    `model_config`::`ClimaAtmos.SphericalModel`
│                       `perf_mode`::`ClimaAtmos.PerfStandard`
│                  `moisture_model`::`ClimaAtmos.EquilMoistModel`
│                     `energy_form`::`ClimaAtmos.TotalEnergy`
│                    `precip_model`::`ClimaAtmos.Microphysics0Moment`
│                    `forcing_type`::`Nothing`
│                      `subsidence`::`Nothing`
│                  `radiation_mode`::`ClimaAtmos.RRTMGPInterface.GrayRadiation`
│                          `ls_adv`::`Nothing`
│                   `edmf_coriolis`::`Nothing`
│                  `turbconv_model`::`Nothing`
│                  `surface_scheme`::`Nothing`
│     `non_orographic_gravity_wave`::`Nothing`
│         `orographic_gravity_wave`::`Nothing`
│                       `hyperdiff`::`ClimaAtmos.ClimaHyperdiffusion{Float64}`
│                       `vert_diff`::`ClimaAtmos.VerticalDiffusion{true, Float64}`
│                  `viscous_sponge`::`Nothing`
│                 `rayleigh_sponge`::`Nothing`
│                 `sfc_temperature`::`ClimaAtmos.ZonallySymmetricSST`
└                   `surface_model`::`ClimaAtmos.PrescribedSurfaceTemperature`
┌ Info: numerics
│   energy_upwinding = Val{:none}()
│   tracer_upwinding = Val{:none}()
│   density_upwinding = Val{:none}()
│   edmfx_upwinding = Val{:none}()
│   apply_limiter = true
└   bubble = true
┌ Info: Time info:
│   dt = "200secs"
│   t_end = "10days"
└   floor_n_steps = 4320
┌ Info: Topography
└   topography = "NoWarp"
┌ Info: z heights
│   z_mesh.faces =
│    11-element Vector{ClimaCore.Geometry.ZPoint{Float64}}:
│     ZPoint(0.0)
│     ZPoint(499.99905386096873)
│     ⋮
└     ZPoint(30000.0)
┌ Info: Resolution stats: 
│   Nq = 4
│   h_elem = 4
│   z_elem = 10
│   ncols = 1536
└   ndofs_total = 15360
[ Info: Allocating Y: 18.325 s (68631626 allocations: 3.30 GiB)
[ Info: Allocating cache (p): 48.058 s (91459588 allocations: 4.14 GiB)
[ Info: Using ODE config: `ClimaTimeSteppers.ARS343`
[ Info: ode_configuration: 884.736 ms (2051508 allocations: 102.51 MiB)
[ Info: get_callbacks: 216.289 ms (162842 allocations: 8.93 MiB)
[ Info: n_steps_per_cycle_per_cb: [108]
[ Info: n_steps_per_cycle: 108
[ Info: Define ode function: 4.677 s (1631444 allocations: 88.38 MiB)
 ```
 
 Do you test interactively in a different way?

Reply:
but we need to define `atmos_config` here